charts/canvas-vault/templates/post-install-hook.yaml

```diff
@@ -0,0 +1,105 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ .Release.Name }}-vault-hc-post-install-hook"
+  namespace: {{ include "canvasvault.namespace" . }}
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+  annotations:
+    # This is what defines this resource as a hook. Without this line, the
+    # job is considered part of the release.
+    "helm.sh/hook": post-install,post-upgrade
+    "helm.sh/hook-weight": "5"
+    "helm.sh/hook-delete-policy": before-hook-creation
+spec:
+  template:
+    metadata:
+      name: "{{ .Release.Name }}-vault-hc-post-install-hook"
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: post-install-job
+        # public image with kubectl and curl:
+        # https://github.com/projectsyn/component-cert-manager/pull/130
+        image: "quay.io/appuio/oc:v4.14"
+        imagePullPolicy: Always
+        command: ["/bin/bash"]
+        args: ["-c", "echo starting canvas vault post install hook;
+           date;
+           set -xev;
+           NAMESPACE={{ include "canvasvault.namespace" . }};
+           VAULTPOD={{ .Release.Name }}-vault-hc-0;
+           AUTH_PATH={{ .Values.auth_path }};
+           JWT=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token);
+           JWT_PL=$(echo $JWT | cut -d '.' -f2);
+           ISSUER=$(echo $JWT_PL==== | base64 -d 2>/dev/null | jq -r '.aud[0]');
+           echo configuring authenticator \"$AUTH_PATH\" for issuer \"$ISSUER\";
+           date;
+           echo waiting up to 120 seconds for the vault to be ready;
+           kubectl -n $NAMESPACE wait -l  statefulset.kubernetes.io/pod-name=$VAULTPOD --for=condition=ready pod --timeout=120s;
+           echo ready;
+           date;
+           echo getting issuer;
+           if [ -z \"$ISSUER\" ];then
+             JSON=$(kubectl exec -n $NAMESPACE -i $VAULTPOD -- sh -c 'wget -O- --no-check-certificate https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_SERVICE_PORT_HTTPS/.well-known/openid-configuration 2>/dev/null');
+             echo ----- response -----;
+             echo $JSON;
+             echo ----- -------- -----;
+             ISSUER=$(echo $JSON | sed -e 's/.*\"issuer\":\"\\([^\"]\\+\\).*/\\1/');
+           fi;
+           echo ISSUER=$ISSUER;
+           if [ -z \"$CACERT\" ]; then
```

Comment:
Please add a short comment, which value CACERT should contain if predefined as env var (or where it otherwise come from)

Reply:
This was copied from the setup shell script and does not work as before. The implemented logic sets the value of CACERT to a fixed path to the mounted Kubernetes root CA cert, or it is empty. other values are not possible.